### DISCUSSION TITLE
Decreasing intrinsic delay after calling Fly-Pie

### DISCUSSION THREAD

**Discussion by D3vil0p3r** [Schneegans/Fly-Pie#231] [Fly-Pie Q&A] [answered]
Hello,
when I test Fly-Pie with the time delay at 0, and I call it  by `CTRL+SPACE`, Fly-Pie is pop up in a time period between 1 and 2 seconds. I know there is always an intrinsic delay in the calls, but my question is: currently Fly-Pie has some degree for decreasing this intrinsic delay for making it more responsive when the user press `CTRL+SPACE`?

And you guys also have this small intrinsic delay between 1 and 2 seconds?

**Comment by Schneegans** [comment]
For me, it appears instantaneously. The delay is definitely below 100ms. You could try to reduce the number of items in your menu and see if it makes any difference. Maybe start with a new menu with just one item... does it open faster? And if you are on X11, you could try switching to Wayland - input processing is much better there.

**Reply by D3vil0p3r** [reply]
> For me, it appears instantaneously. The delay is definitely below 100ms. You could try to reduce the number of items in your menu and see if it makes any difference. Maybe start with a new menu with just one item... does it open faster? And if you are on X11, you could try switching to Wayland - input processing is much better there.

I can confirm that depends on the number of items inside the menu. I already use Wayland. From the point of view of the developing, is it possible to use multithreading approach (or similar methods) in order to elaborate the call of the menu faster in case of a good amount of elements in the menu?

**Reply by Schneegans** [reply]
Another potential source for a delay is the automatic coloring of items based on the average icon color. You could try to tweak your theme to use fixed colors for all items and see if it makes any difference.

Multi-threading is not an option in JavaScript. The loading itself is quite optimized already - many things happen asynchronously. Nevertheless, some things could still be improved, I guess. How many items do you have in your menu?

**Reply by D3vil0p3r** [reply]
> Another potential source for a delay is the automatic coloring of items based on the average icon color. You could try to tweak your theme to use fixed colors for all items and see if it makes any difference.
> 
> Multi-threading is not an option in JavaScript. The loading itself is quite optimized already - many things happen asynchronously. Nevertheless, some things could still be improved, I guess. How many items do you have in your menu?

I'm deploying this menu in a new Linux distro based on Arch Linux and focused on pentesting. This menu can fill well with my purpose because I can break the usual style of classical menu bar (used in the other pentesting OSes) and deploy the security tools across Fly-Pie, that for my scope has the following advantages: 
* touchscreen compatible
* quick access to the tools
* most-used tools well organized
* custom commands for running tools
* graphically cute

The number of elements is around 50 elements (organized in 9 categories/sub-menus), so average we have 5 elements for each category. What could be very nice in my case is a quicker response (despite it is a stupid request, I'm taking the part of the most silly user).

Here there is an example of the menu:

![image](https://user-images.githubusercontent.com/83867734/174447203-af9fe071-eb56-4350-b271-f452c7821de2.png)

and here an example of one of the categories/sub-menus:

![image](https://user-images.githubusercontent.com/83867734/174447238-7875b5da-9880-44b1-97d3-34d8469cb428.png)

**Reply by D3vil0p3r** [reply]
@Schneegans You can find the entire project here: https://github.com/Athena-OS/athena-iso

I quoted your project in Credits section, I hope it is not a problem for you.

**Reply by Schneegans** [reply]
Thanks, that totally fine for me! Another thing which comes to my mind: You should make sure that the item's icons in the menu have a reasonable resolution (e.g. 64x64 pixels). If you choose custom image files with huge resolutions, loading will take much longer.

**Reply by D3vil0p3r** [reply]
Oh. I used much more greater icon size... I try to change the size of the icons and I keep you updated on the result.

**Comment by D3vil0p3r** [comment, marked as answer]
Hello @Schneegans 
As suggested, I resized all the icons. I sized all categories as 256x256 and all the elements inside each category as 128x128 and now Fly-Pie is super responsive, much much better and faster than before. So the issue was the size of the icons. My previous icons were SVG and now I converted them as PNG.
